### PR TITLE
Support Azure blob storage as backend

### PR DESCRIPTION
**Pull request by alikefia**
object_store library seems supporting Azure and the naive implementation is working. Any reason for not supporting it ?

### CODE REVIEW

**Comment by eddyxu** [comment]
Thanks for contributing to Lance, @alikefia 

We are happy to support azure storage. Previously, mostly because we did not get requests for azure storage support, so we have not spent time on it yet.

File: rust/src/io/object_store.rs [hunk 1]
```diff
@@ -139,6 +146,12 @@ impl ObjectStore {
                 base_path: Path::from(url.path()),
                 block_size: 64 * 1024,
             }),
+            "az" => Ok(Self {
```
Comment (eddyxu) [inline rust/src/io/object_store.rs:149]: Should we support all URL schemes described here

https://docs.rs/object_store/latest/object_store/azure/struct.MicrosoftAzureBuilder.html#method.with_url

Comment (alikefia) [reply rust/src/io/object_store.rs:149]: IMO
1/ we should avoid http and https (generic)
2/ abfs ones are mainly for Hadoop
3/ adl is for Gen 1
4/ azure is not on the spec

This why, I restricted it to az.

Comment (eddyxu) [reply rust/src/io/object_store.rs:149]: I see. Thanks for the explanation.

**Review by eddyxu — APPROVED** [review]

**Comment by eddyxu** [comment]
Windows failures are likely due to CI permissions.

**Pull request merged**